core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/DefaultHttpJsonRequest.java

```diff
@@ -202,6 +216,15 @@ protected DefaultHttpJsonResponse doRequest(
     final HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
     conn.setConnectTimeout(timeout > 0 ? timeout : 60000);
     conn.setReadTimeout(timeout > 0 ? timeout : 60000);
+
+    final boolean hasHeaders = headers != null && !headers.isEmpty();
```

Comment:
Can you inline this into if clause?

Reply:
I have decided not to break existing design of the class (see processing of query params above). I can rework both if needed ?

Reply:
you are free to use the style of code you want as long as it doesn't make the code inefficient or too complex (probably empirical for maintainers values ;) ). 
So my comment about this line is just matter of my taste )
Feel free to do as you want here.
And since I approved the PR in the initial state I was OK to merge it at that stage, my comments were just insignificant details.

Reply:
can we initialise ```headers = new ArrayList<>()``` in constructor to not to have ``` if (headers == null) {``` all other the code.

Reply:
@skabashnyuk I'm quite sure that the initial designer of the class who seems to be really keen on performance and even used custom capacity[1] for query list initialization would like this idea. Basically, the rule of rule of thumb for such cases is either do a massive refactoring or follow the design that is already in place e.g [2] (even if you think it is not smth.  you would normally do if implementing from scratch)  


[1] https://github.com/eclipse/che/pull/7898/files#diff-570b40ab3f5e69b575585ffb160ef3b5L115
[2] https://github.com/eclipse/che/pull/7898/files#diff-570b40ab3f5e69b575585ffb160ef3b5R201

Reply:
+1 to what @ibuziuk said